File: 2020/day-1.py
# Day-1: expense report


# Get input from file and add it to array
def getReportList(file):
    with open(file) as f:
        reportList = f.readlines()
    #strip '\n' and convert to int
    reportList = [int(x.strip()) for x in reportList]

    return reportList 

# Day-1_Puzzle-1 Find the two entries that sum to 2020 and then multiply those two numbers together.
def multiplySumOfTwoNumbersOf2020(reportList):
    result = 2020
    list = []
    for i in range (len(reportList)):
        current = reportList[i]
        for j in range (i+1, len(reportList)):
            sum = current + reportList[j]
            if sum == result:
                print('Sum: ' + str(sum) + ' = ' + str(current) + ' + ' + str(reportList[j]))
                print('Product: ' + str(current * reportList[j]) + ' = ' + str(current) + ' * ' + str(reportList[j]))
                return current * reportList[j]

# Day-1_Puzzle-2 Find the three entries that sum to 2020 and then multiply those two numbers together.
def multiplySumOfThreeNumbersOf2020(reportList):
    result = 2020
    for i in range (len(reportList)):
        current = reportList[i]
        for j in range (i+1, len(reportList)):
            for k in range (j+1, len(reportList)):
                sum = current + reportList[j] + reportList[k]
                if sum == result:
                    print('Sum: ' + str(sum) + ' = ' + str(current) + ' + ' + str(reportList[j]) + ' + ' + str(reportList[k]))
                    print('Product: ' + str(current * reportList[j] * reportList[k]) + ' = ' + str(current) + ' * ' + str(reportList[j]) + ' * ' + str(reportList[k]))
                    return (current * reportList[j] * reportList[k])


print()
print(multiplySumOfTwoNumbersOf2020(getReportList('ExpenseReport.txt')))
print()
print(multiplySumOfThreeNumbersOf2020(getReportList('ExpenseReport.txt')))

File: 2020/CSharpAdvent/Day_1.cs
using System;
using System.Collections.Generic;

namespace CSharpAdvent
{
    class Day_1
    {
        public int MultiplySumOfTwoNumbers(List<string> Lines)
        {
            //Day-1-Part-1: Find the two entries that sum to 2020 and then multiply those two numbers together.
            int current = 0;
            int sum = 2020;

            for(int i = 0; i < Lines.Count; i++)
            {
                current = int.Parse(Lines[i]);
                for(int j = 0; j < Lines.Count; j++){
                    int x = int.Parse(Lines[j]);
                    if((current + x) == sum)
                    {
                        System.Console.WriteLine("Current: {0} and x: {1} and sum: {2}", current, x, current + x);
                        return (current * x);
                    }
                }
            } 
            return -1;
        }

        //Day-1-Part-2 Find the three entries that sum to 2020 and then multiply those two numbers together.
        public int MultiplySumOfThreeNumbersOf2020(List<string> Lines)
        {
            int current = 0;
            int sum = 2020;

            for(int i = 0; i < Lines.Count; i++)
            {
                current = int.Parse(Lines[i]);
                for(int j = 0; j < Lines.Count; j++)
                {
                    int x = int.Parse(Lines[j]);
                    for(var k = 0; k < Lines.Count; k++)
                    {
                        int y = int.Parse(Lines[k]);
                        if((current + x + y) == sum)
                        {
                            System.Console.WriteLine("Current: {0} and x: {1} and y: {2} and sum: {3}", current, x, y, current + x);
                            return (current * x * y);
                        }
                    }
                }
            } 
            return -1;   
        }
    }
}